sdk/mobile-center-updates/src/main/java/com/microsoft/azure/mobile/updates/Updates.java

```diff
@@ -0,0 +1,971 @@
+            }
+
+            /* If update dialog was not shown or scheduled, complete workflow. */
+            completeWorkflow();
+        }
+    }
+
+    /**
+     * Check if the fetched release information should be installed.
+     *
+     * @param packageInfo    current app version.
+     * @param releaseDetails latest release on server.
+     * @return true if latest release on server should be used.
+     */
+    private boolean isMoreRecent(PackageInfo packageInfo, ReleaseDetails releaseDetails) {
+//        TODO when releaseHash is exposed in JSON.
+//        if (releaseDetails.getVersion() == packageInfo.versionCode) {
+//            return !releaseDetails.getReleaseHash().equals(computeHash(mContext, packageInfo));
+//        }
+        return releaseDetails.getVersion() > packageInfo.versionCode;
+    }
+
+    /**
+     * Show update dialog. This can be called multiple times if clicking on HOME and app resumed
+     * (it could be resumed in another activity covering the previous one).
+     */
+    private synchronized void showUpdateDialog() {
+
+        /* We could be in another activity now, refresh dialog. */
+        MobileCenterLog.debug(LOG_TAG, "Show update dialog.");
+        if (mUpdateDialog != null) {
+            mUpdateDialog.hide();
+        }
+        AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(mForegroundActivity);
+        dialogBuilder.setTitle(R.string.mobile_center_updates_update_dialog_title);
+        final ReleaseDetails releaseDetails = mReleaseDetails;
+        String releaseNotes = releaseDetails.getReleaseNotes();
+        if (TextUtils.isEmpty(releaseNotes))
+            dialogBuilder.setMessage(R.string.mobile_center_updates_update_dialog_message);
+        else
+            dialogBuilder.setMessage(releaseNotes);
+        dialogBuilder.setPositiveButton(R.string.mobile_center_updates_update_dialog_download, new DialogInterface.OnClickListener() {
+
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                scheduleDownload(releaseDetails);
+            }
+        });
+        dialogBuilder.setNegativeButton(R.string.mobile_center_updates_update_dialog_ignore, new DialogInterface.OnClickListener() {
+
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                completeWorkflow(releaseDetails);
+            }
+        });
+        dialogBuilder.setOnCancelListener(new DialogInterface.OnCancelListener() {
+
+            @Override
+            public void onCancel(DialogInterface dialog) {
+                completeWorkflow(releaseDetails);
+            }
+        });
+        mUpdateDialog = dialogBuilder.create();
+        mUpdateDialog.show();
+    }
+
+    /**
+     * Check state did not change and schedule download of the release.
+     *
+     * @param releaseDetails release details.
+     */
+    private synchronized void scheduleDownload(ReleaseDetails releaseDetails) {
```

Comment:
Anyway it will be changed to `startDownloadOrShowUnknownSourcesDialog` in the PR I am working on.